src/scripts/main.js

```diff
@@ -1,5 +1,32 @@
 'use strict';
 
 document.addEventListener('click', e => {
-  // write code here
+  const spider = document.querySelector('.spider');
+  const wall = document.querySelector('.wall');
+
+  if (e.target !== wall) {
+    return;
+  };
+
+  let x = e.offsetX - spider.height / 2;
+  let y = e.offsetY - spider.width / 2;
+
+  if (x < 0) {
+    x = 0;
+  };
+
+  if (y < 0) {
+    y = 0;
+  }
+
+  if (x > wall.clientHeight - spider.height) {
+    x = wall.clientHeight - spider.height;
+  }
+
+  if (y > wall.clientWidth - spider.width) {
+    y = wall.clientWidth - spider.width;
+  }
```

Comment:
create variables for `wall.clientHeight - spider.height` and `wall.clientWidth - spider.width` as well